content/news/2022-07-29-bevy-0.8/index.md

```diff
@@ -0,0 +1,1131 @@
+
+* **New Material system**: Custom shaders are now _much_ easier to define, thanks to the new Material trait and AsBindGroup derive
+* **Camera driven rendering**: Each Camera now configures what it renders and how it renders it. Easily layer camera renders on top of each other, do split screen, or render to a texture in just a few lines of code.
+* **Built-in Shader Modularization**: Many built in shader types and functions are now importable. Notably, custom shaders can now import the PBR shader logic
+* **Spot Lights**: A new light type that emits light in a cone shape from a fixed point
+* **Visibility Inheritance**: Hiding an entity now also hides all of its descendants in the hierarchy
+* **Scene Bundle**: Easily spawn scenes using a normal Bevy bundle and extend them with new components and children
+* **Upgraded to wgpu 0.13**: Uses a new, more ergonomic WGSL shader syntax
+* **Automatic Mesh tangent generation**: If tangents are missing for a mesh, generate them with mikktspace
+* **Renderer Optimizations**: Parallel Frustum Culling, thread local check visiblity, unstable sorts for unbatched render phases, 
+* **Scripting / Modding Progress: Untyped ECS apis**: A step toward 3rd party scripting language support! Interact with bevy ecs internals directly via pointers.
+* **ECS Query ergonomics and usability**: Queries now implement IntoIter and mutable queries can be converted to immutable queries
+* **ECS internals refactors**: Sweeping changes to Bevy ECS internals that make it simpler, safer, and easier to maintain
+* **Reflection improvements**: Support for reflecting more types, ECS resource reflection, untyped reflection, improved internals
+* **Hierarchy Commands**: Hierarchy updates now use "transactional commands" to ensure hierarchy integrity at a given point in time
+* **Bevy UI now uses Taffy**: Switched to a community maintained fork (that we helped build) of the now abandoned Stretch layout library 
+
+<!-- more -->
+
+## New Material System
+
+<div class="release-feature-authors">authors: @cart, @Wrapperup, @johanhelsing</div>
+
+Bevy has a brand new [`Material`] system that makes defining custom shaders a breeze. Bevy's previous material system required hundreds of lines of "mid-level" rendering api boilerplate. This was never the long term plan, just an intermediate step. In **Bevy 0.8**, custom shader materials are as simple as this:
+
+```rust
+#[derive(AsBindGroup, TypeUuid, Clone)]
+#[uuid = "f690fdae-d598-45ab-8225-97e2a3f056e0"]
+pub struct CoolMaterial {
+    #[uniform(0)]
+    color: Color,
+    #[texture(1)]
+    #[sampler(2)]
+    color_texture: Handle<Image>,
+}
+
+// The Material trait has many optional functions for configuration.
+// In this case, all we need to set is a fragment shader. Ergonomics!
+impl Material for CoolMaterial {
+    fn fragment_shader() -> ShaderRef {
+        "cool_material.wgsl".into()
```

Comment:
One thing I’m wonder is how to do hot-reloading with a custom shader.

Reply:
It still works like before, as long as you insert the AssetServerSetings resource

Reply:
Confirmed. I've been playing with this, and as long as `AssetServerSettings` is inserted before `DefaultPlugins`, hot reloading works great.